cms-contentful/it/ContentfulCmsServiceIT.java

```diff
@@ -27,11 +28,14 @@ public void setUp() throws Exception {
         contentfulCmsService = ContentfulCmsService.of(SPACE_ID, TOKEN);
     }
 
-    @Test(expected=ExecutionException.class)
-    public void whenCouldNotFetchEntry_thenReturnOptionalEmpty() throws Exception {
+    @Test
+    public void whenCouldNotFetchEntry_thenThrowException() throws Exception {
         final ContentfulCmsService cmsService = ContentfulCmsService.of("", "");
         final CmsIdentifier cmsIdentifier = CmsIdentifier.of("entryType:entryKey.fieldName");
-        waitAndGet(cmsService.get(SUPPORTED_LOCALES, cmsIdentifier));
+
+        Throwable thrown = catchThrowable(() -> waitAndGet(cmsService.get(SUPPORTED_LOCALES, cmsIdentifier)));
+
+        assertThat(thrown).isInstanceOf(ExecutionException.class).hasMessageContaining("Unauthorized");
```

Comment:
@dariuszor that's good, but I would add one more assertion with `.hasCauseInstanceOf(X.class)`, being `X` whatever exception Contentful is throwing.
In the short future we have to change this, because otherwise people won't be able to catch this exception (for each connector a different exception will be thrown). Can you create for another PR a `CmsServiceException`? Then wrap the Contentful exception in ours and check here if our exception is the cause.
Put also the missing exception documentation in both `CmsService` and `ContentfulCmsService`.
Do you want me to create a ticket for that, better?